### PR TITLE
Improved news and faq

### PR DESCRIPTION
**Pull request by aderidder**
I've also updated the Customizing-your-instance-of-cBioPortal documentation. Once merged, we can update the cBioPortal wiki with the new version, located here:
https://github.com/thehyve/cbioportal/wiki/Customizing-your-instance-of-cBioPortal

Updates:
* Changed skin.faq to skin.documentation.faq
* Changed skin.news to skin.documentation.news
* Changed skin.about to skin.documentation.about
* Added skin.documentation.baseurl
* Added skin.documentation.markdown
* Small update to "Creating your own news/about/FAQ/or Examples page"
* Added new section "cBioPortal Documentation Settings"

I've also updated the portal.properties.example with these changes.

We should probably add a title to each wiki page, to clarify which page you're on if you're navigating these pages within cBioPortal.

### CODE REVIEW

**Comment by aderidder** [comment]
Hi @adamabeshouse , since this feature has a Spring webservice component, would you be willing to review it? Thanks!

File: core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java [hunk 1]
```diff
@@ -347,6 +363,11 @@ public static String getExamplesRightColumnHtml()
         return (examplesRightColumnHtml == null) ? DEFAULT_SKIN_EXAMPLES_RIGHT_COLUMN : "../../../content/"+examplesRightColumnHtml;
     }
 
+    private static String getContentString(String contentString){
+        if(getBaseUrl().equalsIgnoreCase("")) return "content/"+contentString;
+//        if(!contentString.startsWith("http")) return "content/"+contentString;
```
Comment (adamabeshouse) [inline core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java]: reason to keep this comment?

File: portal/src/main/webapp/js/src/url_based_content.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+        // if target starts with http, open a new tab and navigate to the target
+        if(target.startsWith("http")) window.open(target);
+        // if target starts with a hash, stay on the current page and scroll to the hash
+        else if(target.startsWith("#")){
+            this.scrollToHash();
+        }
+        else{
+            // if the target is internal, check whether the target has a hash; if so, remove it
+            if(this.hash.length>0) target = target.substring(0, target.indexOf('#'));
```
Comment (adamabeshouse) [inline portal/src/main/webapp/js/src/url_based_content.js]: It's my opinion that one line if/else statements are asking for maintenance trouble down the line when you accidentally forget to put in brackets later. Also the irregularity makes it slightly harder to read. But that's mostly taste.

**Comment by adamabeshouse** [comment]
besides small comments/questions :+1: looks good to me

**Comment by inodb** [comment]
Looks good! Just a comment on the commit history. The merge commit 2889f64 seems superfluous. You could use git rebase (http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/) to prevent it. Overall the commits look good. However before merging I think it is nicer to squash the commits into a single commit describing the entire feature, because for most other developers reading the log the small incremental commits are not very informative. If there are commit messages that are essential to certain pieces of code it's better as comments in the code.

**Comment by aderidder** [comment]
@adamabeshouse thanks for reviewing
Absolutely no reason to keep the comment; I'll remove it. I'll also add brackets.
@inodb also thanks for reviewing
Ok interesting. Thanks for the link, I'll read it

**Comment by jjgao** [comment]
@aderidder, just wanted to add to @inodb's point, if there is no conflicts with the targeting branch, you don't have to merge or rebase.

**Comment by aderidder** [comment]
@jj don't I need to merge or rebase to test my code on the new rc?

2015-11-21 21:22 GMT+01:00 JianJiong Gao <notifications@github.com>:

> @aderidder <https://github.com/aderidder>, just wanted to add to @inodb
> <https://github.com/inodb>'s point, if there is no conflicts with the
> targeting branch, you don't have to merge or rebase.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cBioPortal/cbioportal/pull/513#issuecomment-158678769>
> .
>



-- 
E.   sander@thehyve.nl
T.   +31(0)30 700 9713
W.  www.thehyve.nl

the hyve

**Comment by jjgao** [comment]
@aderidder Good point, but I am not sure if it's necessary. Integrative test (testing more than one features) can/will be performed at a later stage when building release candidate.

**Comment by jjgao** [comment]
@aderidder but I don't feel strong about it. Please feel free to rebase.

**Pull request merged**